packages/vite/src/node/optimizer/index.ts

```diff
@@ -156,45 +174,49 @@ export interface DepOptimizationMetadata {
 export async function optimizeDeps(
   config: ResolvedConfig,
   force = config.server.force,
-  asCommand = false,
-  newDeps?: Record<string, OptimizedDepInfo>, // missing imports encountered after server has started
-  ssr?: boolean
+  asCommand = false
 ): Promise<DepOptimizationMetadata> {
-  const { metadata, run } = await createOptimizeDepsRun(
+  const cachedMetadata = loadCachedDepOptimizationMetadata(
     config,
     force,
-    asCommand,
-    null,
-    newDeps,
-    ssr
+    asCommand
   )
-  const result = await run()
+  if (cachedMetadata) {
+    return cachedMetadata
+  }
+  const depsInfo = await discoverProjectDependencies(config)
+
+  const depsString = depsLogString(config, Object.keys(depsInfo))
+  config.logger.info(colors.green(`Optimizing dependencies:\n  ${depsString}`))
```

Comment:
Could you do a code suggestion? I don't get this one

Reply:
Something like this:

```suggestion
  const log = asCommand ? config.logger.info : debug
  log(colors.green(`Optimizing dependencies:\n  ${depsString}`))
```

because we also do so here too:

https://github.com/vitejs/vite/blob/07a01b580dc75429182178822910a934a75eb2df/packages/vite/src/node/optimizer/index.ts#L219